### ISSUE TITLE
Passing model scale for training YOLO12 pose with custom augmentation

### ISSUE THREAD

**Issue by ysuter** [ultralytics/ultralytics#20204]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am trying to train YOLO12-pose models on the coco pose dataset while doing more data augmentation than in the default settings.

I cannot find a way to pass the model scale (e.g., 'm', 'l' etc.) together with the augmentation information.

Here's my training python script:
```
#!/usr/bin/env python3

from ultralytics import YOLO

model = YOLO('yolo12-pose.yaml') 

# Train the model
results = model.train(data='coco.yaml', imgsz=640, device=[0, 1], epochs=1000, batch=64,
                      name='yolo12m-pose-coco_nopretraining-augmented',
                      model='yolo12m-pose.yaml',
                      fliplr=0.5,
                      hsv_h=0.03,
                      hsv_s=0.75,  # (float) image HSV-Saturation augmentation (fraction)
                      hsv_v=0.7,  # (float) image HSV-Value augmentation (fraction)
                      degrees=90.0,  # (float) image rotation (+/- deg)
                      translate=0.1,  # (float) image translation (+/- fraction)
                      scale=0.5,  # (float) image scale (+/- gain)
                      shear=10.0,  # (float) image shear (+/- deg)
                      perspective=0.001,  # (float) image perspective (+/- fraction), range 0-0.001
                      flipud=0.1,  # (float) image flip up-down (probability)
                      bgr=0.1,  # (float) image channel BGR (probability)
                      mosaic=0.5,  # (float) image mosaic (probability)
                      mixup=0.0,  # (float) image mixup (probability)
                      copy_paste=0.0,  # (float) segment copy-paste (probability) # (str) the method to do copy_paste augmentation (flip, mixup)
    )
```

coco.yaml:
```
path: '/PATH_TO_FOLDER/coco-pose'
train: train2017.txt # train images (relative to 'path') 56599 images
val: val2017.txt # val images (relative to 'path') 2346 images
#test: test-dev2017.txt # 20288 of 40670 images, submit to https://codalab.lisn.upsaclay.fr/competitions/7403

# Keypoints
kpt_shape: [17, 3] # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)
flip_idx: [0, 2, 1, 4, 3, 6, 5, 8, 7, 10, 9, 12, 11, 14, 13, 16, 15]

# Classes
names:
  0: person
```

For the yolo12-pose model definition I am using this: [https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/12/yolo12-pose.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/12/yolo12-pose.yaml)

In the file yolo12-pose.yaml, the comment says that the model scale can be selected by passing a corresponding model filename: 

> 'model=yolo12n-pose.yaml' will call yolo12-pose.yaml with scale 'n'

If I do this as described above, I always get the warning "WARNING ⚠️ no model scale passed. Assuming scale='n'." Adding scale='m' as a training argument, that does not work because the augmentation scaling info is expected there.

Can you give me some hints on how to do this correctly? I can comment the unwanted scales in the yolo12-pose.yaml file to use the scale I need, but that seems not to be how that should be working.

Thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ysuter, thank you for your detailed question and for providing your script and config files! 🚀 This is an automated response to help you get the most out of Ultralytics. An Ultralytics engineer will review your issue and assist you further soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for guidance, especially the [Python usage](https://docs.ultralytics.com/usage/python/) and [CLI usage](https://docs.ultralytics.com/usage/cli/) sections, which cover many training scenarios and parameters for YOLO12-pose and other models.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you haven't already, as this helps us debug the issue efficiently.

If your question is about custom training or model scaling, please include as much detail as possible, such as:
- Your full training command or script
- Data configuration
- Model YAML in use
- Full logs, including the exact warning or error message

Make sure you have the latest version of the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Verified Environments

You can also try your workflow in any of our verified environments, each with dependencies pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

A green badge means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests validate YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

If you’d like to discuss with the community, we invite you to join:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussion
- [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge and experiences

Thank you for helping improve Ultralytics!

**Comment by Y-T-G** [comment]
It works fine for me:

```python
In [3]: model = YOLO("yolo12n-pose.yaml")
In [4]: model.info()
YOLO12n-pose summary: 287 layers, 2,886,715 parameters, 2,886,699 gradients, 7.8 GFLOPs
Out[4]: (287, 2886715, 2886699, 7.7554)

In [5]: model = YOLO("yolo12x-pose.yaml")
In [6]: model.info()
YOLO12x-pose summary: 503 layers, 61,134,489 parameters, 61,134,473 gradients, 208.7 GFLOPs
Out[6]: (503, 61134489, 61134473, 208.71718399999997)
```

You should be loading the model like that. Not passing it to `model.train()`.

**Comment by glenn-jocher** [comment]
Glad to hear you got it working with the proper version of Ultralytics! Just a friendly heads-up: while you can certainly continue your experiments with YOLO12-pose, we generally recommend YOLO11-pose models for most use cases as they're more stable during training and significantly more efficient (YOLO12 tends to be 2-3x slower on CPU and requires more memory due to its attention layers). 

If you encounter any training stability issues with your augmented YOLO12-pose model, you might want to try the same augmentation settings with YOLO11-pose models as an alternative. Let us know if you need any further assistance with your training!
